src/runtime_src/core/pcie/common/system_pcie.cpp

```diff
@@ -45,4 +45,40 @@ get_devices(boost::property_tree::ptree& pt) const
   pt.add_child("devices", pt_devices);
 }
 
+void
+system_pcie::
+bdf2index(uint16_t& index, const std::string& bdfStr) const
+{
+  // Extract bdf from bdfStr.
+  int dom = 0, b= 0, d = 0, f = 0;
+  index = std::numeric_limits<uint16_t>::max();
+  char dummy;
+  std::stringstream s(bdfStr);
+  size_t n = std::count(bdfStr.begin(), bdfStr.end(), ':');
+  
+  if (n == 1)
+    s >> std::hex >> b >> dummy >> d >> dummy >> f;
+  else if (n == 2)
+    s >> std::hex >> dom >> dummy >> b >> dummy >> d >> dummy >> f;
+  if ((n != 1 && n != 2) || s.fail()) {
+    std::string errMsg = boost::str( boost::format("Can't extract BDF from '%s'") % bdfStr);
+    throw error(errMsg);
+	}
+
+  for (uint16_t i = 0; i < get_total_devices().first; i++) {
+    auto device = get_mgmtpf_device(i);
+    boost::any bus, dev, func;
+    device->query(xrt_core::device::QR_PCIE_BDF_BUS, typeid(b), bus);
+    device->query(xrt_core::device::QR_PCIE_BDF_DEVICE, typeid(d), dev);
+    device->query(xrt_core::device::QR_PCIE_BDF_FUNCTION, typeid(f), func);
+
+    if (b == boost::any_cast<uint16_t>(bus) && d == boost::any_cast<uint16_t>(dev) && f == boost::any_cast<uint16_t>(func)) {
+      index = i;
+    } else {
+      std::string errMsg = boost::str( boost::format("No mgmt PF found for '%s'") % bdfStr);
+      throw error(errMsg);
```

Comment:
You are throwing out of the for loop for the after just checking first device that doesn't match. Or in other words, you loop handles only one device